rajawali/src/main/java/c/org/rajawali3d/core/AFrameDelegate.java

```diff
@@ -0,0 +1,234 @@
+                if (callback.callFrameStart()) {
+                    frameStartCallbacks.add(callback);
+                }
+                if (callback.callFrameEnd()) {
+                    frameEndCallbacks.add(callback);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeFrameCallback(@NonNull final FrameCallback callback) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                if (callback.callFrameStart()) {
+                    frameStartCallbacks.remove(callback);
+                }
+                if (callback.callFrameEnd()) {
+                    frameEndCallbacks.remove(callback);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean clearFrameCallbacks() {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                frameStartCallbacks.clear();
+                frameEndCallbacks.clear();
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean addAnimation(final Animation anim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.add(anim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeAnimation(final Animation anim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.remove(anim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean replaceAnimation(final Animation oldAnim, final Animation newAnim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.set(animations.indexOf(oldAnim), newAnim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean addAnimations(final Collection<Animation> anims) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.addAll(anims);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean clearAnimations() {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.clear();
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    //
+    // ACoreComponent method overrides
+    //
+
+    @RenderThread
+    @Override
+    @CallSuper
+    public void onFrameStart(double deltaTime) throws InterruptedException {
+        if (!isEnabled) {
```

Comment:
I guess it's like most such issues, a trade-off between concerns.  I see that it would be easier to understand stack traces and control flows if you can assume super-first, but avoiding overriding of super methods whenever that convention might be violated seems limiting too. I don't really get the "natural order" argument, except for constructors. Don't C++ destructors always run super-last as the "natural order"? In between, it seems arbitrary to insist on super first.  

For example, in my app's Android activity life cycle, I need to deal with model initialization before fragments get their onStarts delegated by the Android parent class, and to disable model-observer notifications before allowing fragments to get onPause events. And in composite-patterns, it seem awkward to hold child container classes (which might form a subtree themselves) responsible to delegate the nested child iteration; instead, it gets confusing about which node is responsible for what, and/or the logic gets duplicated. 

In this case, I think I'd prefer to just check for enabled in the caller. That avoids your concern here and we can debate the rest of it over a beer some day. Wait, no, that could ruin a really good quaft...